src/Widgets/NotificationEntry.vala

```diff
@@ -186,6 +193,33 @@ public class Notifications.NotificationEntry : Gtk.ListBoxRow {
 
         grid.attach (body_label, 1, 1, 2);
 
+        if (notification.actions_with_label.length > 0) {
+            var action_area = new Gtk.Box (Gtk.Orientation.HORIZONTAL, 6) {
+                margin_top = 12,
+                halign = Gtk.Align.END,
+                homogeneous = true
+            };
+
+            bool action_area_packed = false;
+            foreach (var action_name in notification.actions_with_label.get_keys ()) {
+                var button = new Gtk.Button.with_label (notification.actions_with_label[action_name]) {
+                    action_name = ACTION_PREFIX + action_name
+                };
+
+                button.clicked.connect (() => {
+                    activate ();
+                    clear (false);
+                });
```

Comment:
you don't need this, you already set the `action_name` in the button.

Reply:
But we need this to close the popover and remove the notification don't we?

Reply:
i expect the bubble to be removed when the server emit ActionInvoked. you need to deal with this case anyway, otherwise bubbles that was already activated before the bubble fade-out will be kept in the indicator.

Reply:
Oh right I totally forgot about that, thanks for the reminder :)